generators/generator-base.js

```diff
@@ -1438,14 +1439,21 @@ module.exports = class extends Generator {
             return e1.definition.changelogDate - e2.definition.changelogDate;
         }
 
-        if (shelljs.test('-d', JHIPSTER_CONFIG_DIR)) {
-            shelljs.ls(path.join(JHIPSTER_CONFIG_DIR, '*.json')).forEach((file) => {
-                const definition = this.fs.readJSON(file);
-                entities.push({ name: path.basename(file, '.json'), definition });
-            });
+        if (!shelljs.test('-d', JHIPSTER_CONFIG_DIR)) {
+            return entities;
         }
 
-        return entities.sort(isBefore);
+        return shelljs.ls(
+            path.join(JHIPSTER_CONFIG_DIR, '*.json')
+        ).reduce((acc, file) => {
+            try {
+                const definition = jhiCore.readEntityJSON(file);
+                acc.push({ name: path.basename(file, '.json'), definition });
+            } catch (error) {
+                // not an entity file / malformed?
```

Comment:
this is based on [jhipster-core lib/reader/json_reader.js#51](https://github.com/jhipster/jhipster-core/blob/master/lib/reader/json_reader.js#L51) which doesn't log at the moment so I assumed there could be other files / directories there that could trigger it too frequently